src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs

```diff
@@ -412,11 +404,10 @@ public static AffineColumnFunction Create(ModelLoadContext ctx, IHost host, Colu
                 throw host.ExceptUserArg(nameof(AffineArgumentsBase.Column), "Wrong column type. Expected: R4, R8, Vec<R4, n> or Vec<R8, n>. Got: {0}.", typeSrc.ToString());
             }
 
-            private abstract class ImplOne<TFloat> : AffineColumnFunction, NormalizerTransformer.IAffineData<TFloat>
+            public abstract class ImplOne<TFloat> : AffineColumnFunction, NormalizerTransformer.IAffineData<TFloat>
```

Comment:
>IAffineData [](start = 96, length = 11)

what's the point of these interfaces?

Reply:
They are conveniences Tom created for the static API, i think mainly to categorize OnFit delegates based on normalization type. Conveniently re-using for the weight exposures . 

---
In reply to: [232814715](https://github.com/dotnet/machinelearning/pull/1594#discussion_r232814715) [](ancestors = 232814715)

Reply:
I don't think it is very convenient, to be honest :)

---
In reply to: [233191036](https://github.com/dotnet/machinelearning/pull/1594#discussion_r233191036) [](ancestors = 233191036,232814715)

Reply:
I'd prefer not to use it in the dynamic API. And maybe also change the static API not to use it either, once we have a dynamic way of accessing things.

---
In reply to: [233685441](https://github.com/dotnet/machinelearning/pull/1594#discussion_r233685441) [](ancestors = 233685441,233191036,232814715)